tools/c7n_azure/c7n_azure/container_host/host.py

```diff
@@ -0,0 +1,415 @@
+                               seconds=queue_poll_seconds,
+                               id="poll_queue")
+
+        self.scheduler.start()
+
+    def update_policies(self):
+        """
+        Enumerate all policies from storage.
+        Use the MD5 hashes in the enumerated policies
+        and a local dictionary to decide if we should
+        bother downloading/updating each blob.
+        We maintain an on-disk policy cache for future
+        features.
+        """
+        if not self.policy_blob_client:
+            self.policy_blob_client = Storage.get_blob_client_by_uri(self.policy_storage_uri,
+                                                                     self.session)
+        (client, container, prefix) = self.policy_blob_client
+
+        try:
+            # All blobs with YAML extension
+            blobs = [b for b in client.list_blobs(container) if Host.has_yaml_ext(b.name)]
+        except AzureHttpError as e:
+            # If blob methods are failing don't keep
+            # a cached client
+            self.policy_blob_client = None
+            raise e
+
+        # Filter to hashes we have not seen before
+        new_blobs = [b for b in blobs
+                     if b.properties.content_settings.content_md5 != self.blob_cache.get(b.name)]
+
+        # Get all YAML files on disk that are no longer in blob storage
+        cached_policy_files = [f for f in os.listdir(self.policy_cache)
+                               if Host.has_yaml_ext(f)]
+
+        removed_files = [f for f in cached_policy_files if f not in [b.name for b in blobs]]
+
+        if not (removed_files or new_blobs):
+            return
+
+        # Update a copy so we don't interfere with
+        # iterations on other threads
+        policies_copy = self.policies.copy()
+
+        for f in removed_files:
+            path = os.path.join(self.policy_cache, f)
+            self.unload_policy_file(path, policies_copy)
+
+        # Get updated YML files
+        for blob in new_blobs:
+            policy_path = os.path.join(self.policy_cache, blob.name)
+            if os.path.exists(policy_path):
+                self.unload_policy_file(policy_path, policies_copy)
+
+            client.get_blob_to_path(container, blob.name, policy_path)
+            self.load_policy(policy_path, policies_copy)
+            self.blob_cache.update({blob.name: blob.properties.content_settings.content_md5})
+
+        # Assign our copy back over the original
+        self.policies = policies_copy
+
+        if self.require_event_update:
+            self.update_event_subscriptions()
+
+    def load_policy(self, path, policies):
+        """
+        Loads a YAML file and prompts scheduling updates
+        :param path: Path to YAML file on disk
+        :param policies: Dictionary of policies to update
+        """
+        with open(path, "r") as stream:
+            try:
+                policy_config = yaml.safe_load(stream)
```

Comment:
using c7n.policy.load here would be preferrable as it will get policy initialization and validation done as well, and then just handle the merge with the resulting collection policy.